lib/resources.js

```diff
@@ -449,10 +449,25 @@ module.exports = function (app, _private = null) {
    * @returns {object}
    */
   const esRangeValue = (range) => {
-    return {
-      gte: range[0],
-      lte: range[range.length - 1]
+    // the greater-than-equal value will always be the first value in the range array.
+    // depending on the number of values and their equality, we query using less-than-equal
+    // the second value, or just less-than the first value plus one
+    let dateRangeQuery = {
+      gte: range[0]
+    }
+    if (range.length === 2) {
+      // if the two range values are the same, query up until the next year
+      if (range[0] === range[1]) {
+        dateRangeQuery.lt = range[0] + 1
+      } else {
+        // otherwise search on both range values
+        dateRangeQuery.lte = range[range.length - 1]
+      }
```

Comment:
This is correct but might be a little easier to follow if we first normalize out the case where range[0] === range[1] so we don't have to repeat the single-value case:
```
// Treat case where range start equals range end same as case of single value:
if (range[0] === range[1]) range = range.slice(0, 1)

// Handle 2-value range:
if (range.length === 2) {
  dateRangeQuery.lte = range[range.length - 1]
} else {
  // if there is just one range value, query up until the next year
  dateRangeQuery.lt = range[0] + 1
}